index.js

```diff
@@ -66,11 +68,55 @@ const eventValidator = new Schema({
         length: {min : 3, max : 255}
     },
     deadline: {
-        type: Date,
+        type: String,
+        match: /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/,
+        required: true
+    }
+})
+
+const participantValidator = new Schema({
```

Comment:
This comment isn't specifically related to this line, but I'd say this file is starting to get a bit big now - would suggest splitting it out into some separate files.

No need to go full controller service (unless you want to ofc), at a minimum I'd suggest putting your validators into a validators folder, and putting your DB middlewares into a middleware folder